jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/PushApplicationEndpoint.java

```diff
@@ -70,27 +71,34 @@
      *
      * @statuscode 201 The PushApplication Variant created successfully
      * @statuscode 400 The format of the client request was incorrect
+     * @statuscode 409 The PushApplication already exists
      */
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @ReturnType("org.jboss.aerogear.unifiedpush.api.PushApplication")
     public Response registerPushApplication(PushApplication pushApp) {
 
-         // some validation
         try {
             validateModelClass(pushApp);
         } catch (ConstraintViolationException cve) {
+            return createBadRequestResponse(cve.getConstraintViolations()).build();
+        }
 
-            // Build and return the 400 (Bad Request) response
-            ResponseBuilder builder = createBadRequestResponse(cve.getConstraintViolations());
-
-            return builder.build();
+        try {
+            pushAppService.addPushApplication(pushApp);
+        } catch (IllegalArgumentException e) {
+            return Response.status(Status.CONFLICT).entity(e.getMessage()).build();
         }
 
-        pushAppService.addPushApplication(pushApp);
+        URI uri = UriBuilder
```

Comment:
let's get the `URI` inline

Reply:
Don't you think it's a bit long code to be inline?
```
return Response.created(UriBuilder.fromResource(PushApplicationEndpoint.class)
                .path(pushApp.getPushApplicationID())
                .build())
                .entity(pushApp)
                .build();
```

Reply:
@josemigallas Nope. If you assign it to a variable, when I read the code I think you will use the variable somewhere else too. If you inline it, I automatically understand it won't be used in any other place.